x-pack/plugins/observability_solution/profiling/server/routes/apm.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema, TypeOf } from '@kbn/config-schema';
+import { termQuery } from '@kbn/observability-plugin/server';
+import { keyBy } from 'lodash';
+import { IDLE_SOCKET_TIMEOUT, RouteRegisterParameters } from '.';
+import { getRoutePaths } from '../../common';
+import { handleRouteHandlerError } from '../utils/handle_route_error_handler';
+import { getClient } from './compat';
+
+const querySchema = schema.object({
+  timeFrom: schema.number(),
+  timeTo: schema.number(),
+  functionName: schema.string(),
+  serviceNames: schema.arrayOf(schema.string()),
+});
+
+type QuerySchemaType = TypeOf<typeof querySchema>;
+
+export function registerTopNFunctionsAPMTransactionsRoute({
+  router,
+  logger,
+  dependencies: {
+    start: { profilingDataAccess },
+    setup: { apmDataAccess },
+  },
+}: RouteRegisterParameters) {
+  const paths = getRoutePaths();
+  router.get(
+    {
+      path: paths.APMTransactions,
```

Comment:
nit: it could be just me but I find the code easier to read when the url is defined here as `${BASE_ROUTE_PATH}/topn/functions/apm/transactions`

Reply:
Me too actually, but we use this for now because TS does not automatically infer the APIs in the frontend.